<!-- This launch file start the detected obstacle reaction distance (DORD) calculator.  This node determines how far away the robot is from an obstacle
     when the robot detects the new obstacle -->

<launch>
    <!-- Save the path for the logs using the current time -->
    <arg name="robot" />
    <arg name="level" />
    <param name="log_file_path" command="bash -c 'printf &quot;$(find uml_hri_nerve_navigation)/resources/logs/$(arg level)_$(arg robot)_%(%d_%m_%Y_%I_%M_%S)T&quot;'" />

    <!-- Get environment info -->
    <arg name="goal" default="goal_b" />
    <arg name="obstacle_x" default="0" />
    <arg name="obstacle_y" default="0" />

    <!-- Start the dord calculator-->
    <node name="dord_calculator" type="dord_calculator" pkg="uml_hri_nerve_navigation" output="screen">
        <remap from="~file_path" to="log_file_path" />
        <param name="goal_name" value="$(arg goal)"/>
        <param name="obstacle_x" value="$(arg obstacle_x)" />
        <param name="obstacle_y" value="$(arg obstacle_y)" />
    </node>

</launch>